src/System.Private.Xml/tests/XmlReader/Tests/ReaderEncodingTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class ReaderEncodingTests
+    {
+        [Fact]
+        public static void ReadWithSurrogateCharAndInvalidChar()
+        {
+            var bytes = new byte[] { 60, 0, 0, 0, 0, 34, 1, 0, 62, 100, 60, 47, 97, 62, 10 };
+            var reader = XmlReader.Create(new MemoryStream(bytes));
+
+            XmlException ex = Assert.Throws<XmlException>(() => reader.Read());
+            Assert.Equal("Invalid character in the given encoding. Line 1, position 4.", ex.Message);
```

Comment:
this won't work when using non en-US locale (although I think we currently don't support it). I'd just check if contains " 1" and " 4" perhaps or not check it at all

Reply:
Yeah I wasn't so confident with asserting error messages. Must be not consistent all time, actually we don't need these assertions, point of the test is there is no Error/Exception while decoding the bytes but expect XmlException while parsing to xml structure, so might be better to remove these assertions.

Reply:
ok, in such cases please also add test cases where invalid surrogates can be found somewhere in between of valid xml

Reply:
have that scenario here: https://github.com/buyaa-n/corefx/pull/1/files#diff-47e5e9b8251940e29f847159c2fd2fb4R55